### PR TITLE
Refactored DOS

### PR DESCRIPTION
**Pull request by elicn**
Changelog highlights:
- Turned the monolithic implementation into a modular one by extracting the interrupt handlers out to dedicated modules
- Broke interrupt handlers down to leaf functions for better readability
- Implemented DOS header parsing for EXE files (as opposed to naked COM files)
- Cleaned and tidied code up
- Refreshed DOS tests

### CODE REVIEW

File: tests/test_dos.py [hunk 1]
```diff
@@ -7,12 +7,32 @@
...
+class Checklist:
+    def __init__(self) -> None:
+        self.visited_onenter = False
+        self.visited_onexit = False
 
 class DOSTest(unittest.TestCase):
 
     def test_dos_8086_hello(self):
-        ql = Qiling(["../examples/rootfs/8086/dos/HI.DOS_COM"], "../examples/rootfs/8086/dos")
+        ql = Qiling(["../examples/rootfs/8086/dos/HI.DOS_COM"], "../examples/rootfs/8086/dos", output='debug')
```
Comment (xwings) [inline tests/test_dos.py:20]: i am thiniking we should remove output="" for good and keep only verbose ?

Comment (elicn) [reply tests/test_dos.py:20]: I always found `output` and `verbose` a bit confusing, but in the current impl both are needed.
Personally, I use `output='debug'` only when I need to enable assembly-level tracing*. We should think of a way to consolidate them both into one variable that would reflect all needed verbosity levels and allow tracing.

( * see what I did for `hello_x8664_linux_disasm.py` )

Comment (xwings) [reply tests/test_dos.py:20]: i agree. I will merge this and we need to do it next.

**Comment by wtdcode** [comment]
For EXE files, PSP also needs to be implemented.

**Pull request merged**